cli/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"math"
+	"os"
+	"strconv"
+	"time"
+
+	"context"
+
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/onchain"
+	"github.com/attestantio/go-eth2-client/spec"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
+)
+
+type ValidatorWithIndex = struct {
+	Validator *phase0.Validator
+	Index     uint64
+}
+
+func main() {
+	eigenpod := flag.String("eigenpod", "", "[required] The onchain address of your eigenpod contract (0x123123123123)")
+	beacon := flag.String("beacon", "", "[required] URI to a functioning beacon node RPC (https://)")
+	node := flag.String("node", "", "[required] URI to a functioning execution-layer RPC")
+	out := flag.String("output", "", "Output path for the proof. (defaults to stdout)")
+	help := flag.Bool("help", false, "Prints the help message and exits.")
+
+	flag.Parse()
+
+	if help != nil && *help {
+		flag.Usage()
+		log.Fatal("Showing help.")
+	}
+
+	if *eigenpod == "" || *beacon == "" || *node == "" {
+		flag.Usage()
+		log.Fatal("Must specify: -eigenpod, -beacon, and -node.")
+	}
+
+	ctx := context.Background()
+
+	execute(ctx, *eigenpod, *beacon, *node, out)
+}
+
+func getBeaconClient(beacon_uri string) (BeaconClient, error) {
+	beaconClient, _, err := NewBeaconClient(beacon_uri)
+	return beaconClient, err
+}
+
+func lastCheckpointedForEigenpod(eigenpodAddress string, client *ethclient.Client) uint64 {
+	eigenPod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), client)
+	PanicOnError(err)
+
+	timestamp, err := eigenPod.LastCheckpointTimestamp(nil)
```

Comment:
you want to get the `currentCheckpointTimestamp` if you're assuming they call this after creating a checkpoint

Reply:
oh sweet good catch 👍